src/__tests__/components/Navbar.test.tsx

```diff
@@ -3,23 +3,59 @@ import '../../__mocks__/matchMediaMock'
 import { Navbar as HospitalRunNavbar } from '@hospitalrun/components'
 import { mount } from 'enzyme'
 import { createMemoryHistory } from 'history'
+import _ from 'lodash'
 import React from 'react'
 import { act } from 'react-dom/test-utils'
+import { Provider } from 'react-redux'
 import { Router } from 'react-router-dom'
+import createMockStore from 'redux-mock-store'
+import thunk from 'redux-thunk'
 
 import Navbar from '../../components/Navbar'
+import Permissions from '../../model/Permissions'
+import { RootState } from '../../store'
+
+const mockStore = createMockStore<RootState, any>([thunk])
 
 describe('Navbar', () => {
   const history = createMemoryHistory()
-  const setup = () =>
-    mount(
+
+  const setup = (permissions: Permissions[]) => {
+    const store = mockStore({
+      title: '',
+      user: { permissions },
+    } as any)
+
+    const wrapper = mount(
       <Router history={history}>
-        <Navbar />
+        <Provider store={store}>
+          <Navbar />
+        </Provider>
       </Router>,
     )
-
-  const wrapper = setup()
+    return wrapper
+  }
+
+  const allPermissions = [
+    Permissions.ReadPatients,
+    Permissions.WritePatients,
+    Permissions.ReadAppointments,
+    Permissions.WriteAppointments,
+    Permissions.DeleteAppointment,
+    Permissions.AddAllergy,
+    Permissions.AddDiagnosis,
+    Permissions.RequestLab,
+    Permissions.CancelLab,
+    Permissions.CompleteLab,
+    Permissions.ViewLab,
+    Permissions.ViewLabs,
+    Permissions.ViewIncidents,
+    Permissions.ViewIncident,
+    Permissions.ReportIncident,
+  ]
+  const wrapper = setup(allPermissions)
```

Comment:
I think it makes more sense to do `const wrapper = setup(allPermissions)` in each `it` test. That way each test doesn't "share" a wrapper.

Reply:
✅ in 860fb8c 🎈